### DISCUSSION TITLE
[UX] annotate tool, replace dropdown menu by search bar

### DISCUSSION THREAD

**Discussion by arthurlutz** [tosdr/edit.tosdr.org#1001] [Ideas] [answered]
In the annotate tool, for those who don't know the list of cases by heart (which is my case as a new comer), it would be nice to have a search tool bar instead of an html drop down. For example when you know the highlighted text has something to do with "delete" it would be really faster to search for delete to see the points concerned.

**Comment by madoleary** [comment]
I think this is a good idea. I agree that the drop-down menu can be overwhelming, but I also believe there's value in users being able to see all of their options so that they choose the correct case. I think, in the long term, a good number of cases could and should be consolidated because there's a lot of overlap between them. @michielbdejong and @Vinnl -- what do you think about this search bar idea for the annotations?

**Comment by arthurlutz** [comment]
@madoleary sorry, my description was confusing. I was thinking of a list with a filter search bar on top. A quick search led me to https://kilianvalkhof.com/uploads/listfilter/ (which is technically described in https://kilianvalkhof.com/2010/javascript/how-to-build-a-fast-simple-list-filter-with-jquery/) This is from 2010 so there are probably better ways of doing this, I'll leave the frontend developers take a look at this.

**Comment by Vinnl** [comment]
I agree that a combination of the two as described by Arthur would probably be useful. For me, to, my workflow is usually to open the list of cases in a separate tab to find the one to use, which is clearly not ideal. Thus, in my view, the main blocker here would be development effort :)

**Comment by madoleary** [comment]
Hi, I've been working on this. I deployed something that at least orders the topics by frequency of use. However, implementing the search bar has been more difficult. I played with this gem: https://github.com/argerim/select2-rails. It creates a searchable drop down menu. But it does not interact well at all with the current javascript on the annotate page. The "toast" component is one issue. Is there not another type of component that we could use to store the drop down menu?

**Comment by augusto-herrmann** [comment]
> For me, to, my workflow is usually to open the list of cases in a separate tab to find the one to use, which is clearly not ideal.

I, too, do exactly the same thing! It makes it much easier to find the case I'm looking for, but a list with a filter would indeed be much better.

**Comment by JCBerger** [comment]
I seem to remember hearing of a page or document somewhere, but is there a place where all of the cases are listed out?

**Reply by JustinBack** [reply]
We do have a list of cases! https://edit.tosdr.org/cases

**Comment by JustinBack** [comment, marked as answer]
I have added this to our new roadmap https://git.tosdr.org/tosdr/roadmap/issues/4
